dali/test/python/test_operator_decoders_image.py

```diff
@@ -90,26 +88,124 @@ def log(img_type, size, device, threads):
                     run_decode(data_path, batch_size, device, threads)
                     yield log, img_type, batch_size, device, threads
 
+# TODO(januszl): check padding behavior
+@pipeline_def
+def create_decoder_slice_pipeline(data_path, device):
+    jpegs, _ = fn.readers.file(file_root = data_path,
+                               shard_id = 0,
+                               num_shards = 1,
+                               name = "Reader")
 
-class DecoderPipelineFastIDC(Pipeline):
-    def __init__(self, data_path, batch_size, num_threads, use_fast_idct=False):
-        super(DecoderPipelineFastIDC, self).__init__(batch_size, num_threads, 0, prefetch_queue_depth=1)
-        self.input = ops.readers.File(file_root = data_path,
-                                      shard_id = 0,
-                                      num_shards = 1)
-        self.decode = ops.decoders.Image(device = 'cpu', output_type = types.RGB, use_fast_idct=use_fast_idct)
+    anchor = fn.random.uniform(range=[0.05, 0.15], shape=(2,))
+    shape = fn.random.uniform(range=[0.5, 0.7], shape=(2,))
+    # ToDo make window random?
+    images_sliced_1 = fn.decoders.image_slice(jpegs,
+                                              anchor,
+                                              shape,
+                                              device = device,
+                                              hw_decoder_load = 0.7,
+                                              output_type = types.RGB,
+                                              axes = (0, 1))
 
-    def define_graph(self):
-        inputs, labels = self.input(name="Reader")
-        output = self.decode(inputs)
-        return (output, labels)
+    images = fn.decoders.image(jpegs,
+                               device = device,
+                               hw_decoder_load = 0.7,
+                               output_type = types.RGB)
+    # ToDo make window random?
+    images_sliced_2 = fn.slice(images,
+                               anchor,
+                               shape,
+                               axes = (0, 1))
+
+    return images_sliced_1, images_sliced_2
+
+# ToDo - check padding behavior
+@pipeline_def
+def create_decoder_crop_pipeline(data_path, device):
+    jpegs, _ = fn.readers.file(file_root = data_path,
+                               shard_id = 0,
+                               num_shards = 1,
+                               name = "Reader")
+
+    # ToDo make window random?
```

Comment:
If you wanted to make this random you can use `fn.random.uniform` for crop_pos_x and crop_pos_y

Reply:
Done